### ISSUE TITLE
linter: false positive for `FunctionExpression`s in `unicorn/consistent-function-scoping`

### ISSUE THREAD

**Issue by DonIsaac** [oxc-project/oxc#5365]
This currently passes, but it should not:
```js
function outer() {
  const inner = function inner() {}
}
```

See [this thread](https://github.com/oxc-project/oxc/pull/5312#discussion_r1735535252) for details.

**Comment by camc314** [comment]
wait i'm confused 😅 

to be clear, the correct behavour here is that the following code:


```ts
function outer() {
  const inner = function inner() {}
}
```

should report an error (`inner` can be moved to the outer scope)

but with oxlint's current behaviour, it does not report an error?

---

if so, isn't this fixed by https://github.com/oxc-project/oxc/pull/5312, with the new test case here ? https://github.com/oxc-project/oxc/pull/5312/files#diff-8becea68b83c5a44bd998989f16fb08f3fc78f8ccfbbd95a30cc3056a6cf0edc

**Comment by DonIsaac** [comment]
<img width="757" alt="image" src="https://github.com/user-attachments/assets/f72334bb-fad6-41ea-a8fe-07ffa0fceb4c">

**Comment by camc314** [comment]
ahhh i see - thanks

In summary:

we do not report an error for the following code:

```ts
let inner;

function outer() {
  inner = function inner() {}
}
```

But we should as there's no reason why the assignment couldn't be removed to the outer scope.


NOTE: when/if we implement this, we should add the following as a PASS test case:
```ts
let inner;

function foo1() {
    inner = function {}
}
function foo2() {
    inner = function {}
}
```

I can't think of why someone would want to do this (maybe a different logger in dev vs prod or something?), but i don't think we should report this case

**Comment by Arian94** [comment]
@camc314 would work on this.

**Comment by Arian94** [comment]
> @camc314 would work on this.

@camc314 I mean myself 🍡. Assign it to me if you'd want.
